src/jdk.jshell/share/classes/jdk/internal/jshell/tool/Startup.java

```diff
@@ -329,18 +336,20 @@ static Startup noStartup() {
      * @param mh handler for error messages
      * @return The default Startup, or empty startup when error (message has been printed)
      */
-    static Startup defaultStartup(boolean preview, MessageHandler mh) {
-        int idx = preview ? 1 : 0;
+    static Startup defaultStartup(int sourceLevel, boolean preview, MessageHandler mh) {
+        boolean startup24 = sourceLevel < 25;
```

Comment:
should we use a constant here?

Reply:
Yes, there should be a constant. Done here:
https://github.com/openjdk/jdk/pull/23801/commits/d66703c1cb4f9f532f9a98181b598fa45e55ef73
Thanks!